source/plugins/ruby/in_kube_podinventory.rb

```diff
@@ -122,275 +163,158 @@ def enumerate(podList = nil)
         if telemetryFlush == true
           telemetryProperties = {}
           telemetryProperties["Computer"] = @@hostName
+          telemetryProperties["PODS_CHUNK_SIZE"] = @PODS_CHUNK_SIZE
+          telemetryProperties["PODS_EMIT_STREAM_BATCH_SIZE"] = @PODS_EMIT_STREAM_BATCH_SIZE
           ApplicationInsightsUtility.sendCustomEvent("KubePodInventoryHeartBeatEvent", telemetryProperties)
           ApplicationInsightsUtility.sendMetricTelemetry("PodCount", @podCount, {})
+          ApplicationInsightsUtility.sendMetricTelemetry("ServiceCount", @serviceCount, {})
           telemetryProperties["ControllerData"] = @controllerData.to_json
           ApplicationInsightsUtility.sendMetricTelemetry("ControllerCount", @controllerSet.length, telemetryProperties)
           if @winContainerCount > 0
             telemetryProperties["ClusterWideWindowsContainersCount"] = @winContainerCount
             ApplicationInsightsUtility.sendCustomEvent("WindowsContainerInventoryEvent", telemetryProperties)
           end
+          ApplicationInsightsUtility.sendMetricTelemetry("PodInventoryE2EProcessingLatencyMs", @podInventoryE2EProcessingLatencyMs, telemetryProperties)
```

Comment:
what is the use of these 2 metrics as you are just getting the last one for the last chunk ? What we need is keep track of an agg (like max,p95) during the telemetryFlush interval, and send send that as agg for that interval.

Reply:
> what is the use of these 2 metrics as you are just getting the last one for the last chunk ? What we need is keep track of an agg (like max,p95) during the telemetryFlush interval, and send send that as agg for that interval.

> similar to other metrics, just emitting at the flush intervals and please note, these metrics are emitted not just last chunk rather accumulated values of the all the chunks. This will help us to understand our ingestion latency where we spending.